kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/BaseAsset.java

```diff
@@ -320,27 +339,41 @@ public ComponentConfiguration getConfiguration() throws KuraException {
             if (descriptor instanceof List<?>) {
                 driverSpecificChannelConfiguration = (List<Tad>) descriptor;
             }
-            if (driverSpecificChannelConfiguration != null) {
-                final ChannelDescriptor basicChanneldescriptor = new BaseChannelDescriptor();
-                List<Tad> channelConfiguration = null;
-                final Object baseChannelDescriptor = basicChanneldescriptor.getDescriptor();
-                if (baseChannelDescriptor instanceof List<?>) {
-                    channelConfiguration = (List<Tad>) baseChannelDescriptor;
-                }
-                if (channelConfiguration != null) {
-                    channelConfiguration.addAll(driverSpecificChannelConfiguration);
-                }
-                for (final Tad attribute : channelConfiguration) {
-                    final Set<String> channelPrefixes = retrieveChannelPrefixes(
-                            this.assetConfiguration.getAssetChannels());
-                    for (final String prefix : channelPrefixes) {
-                        final Tad newAttribute = cloneAd(attribute, prefix);
-                        mainOcd.addAD(newAttribute);
-                    }
+
+            fillDriverSpecificChannelConfiguration(mainOcd, driverSpecificChannelConfiguration);
+        }
+        return new ComponentConfigurationImpl(componentName, mainOcd, props);
+    }
+
+    /**
+     * Fills the {@code mainOcd} with the driver specific configuration provided by
+     * {@code driverSpecificChannelConfiguration}
+     * 
+     * @param mainOcd
+     *            a {@link Tocd} object.
+     * @param driverSpecificChannelConfiguration
+     *            the driver specific configuration.
+     */
+    @SuppressWarnings("unchecked")
+    private void fillDriverSpecificChannelConfiguration(final Tocd mainOcd,
+            final List<Tad> driverSpecificChannelConfiguration) {
+        if (mainOcd == null || driverSpecificChannelConfiguration == null) {
```

Comment:
This checks reduces the unnecessary variable references creation that happens after this.